src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -45,14 +45,7 @@ internal async ValueTask<DValue<T>> EvalArgAsync<T>(FormulaValue arg, SymbolCont
         {
             if (arg is LambdaFormulaValue lambda)
             {
-                var val = await lambda.EvalAsync(this, context);
-                return val switch
-                {
-                    T t => DValue<T>.Of(t),
-                    BlankValue b => DValue<T>.Of(b),
-                    ErrorValue e => DValue<T>.Of(e),
-                    _ => DValue<T>.Of(CommonErrors.RuntimeTypeMismatch(irContext))
```

Comment:
When OptionSetValue derived from FormulaValue directly, it would fall into this error case. 
Change it to derive from ValidFormulaValue so that it is a success path.